src/gui/MainWindow.cpp

```diff
@@ -1497,43 +1497,32 @@ void MainWindow::fillTemplatesMenu()
 {
 	m_templatesMenu->clear();
 
-	QDir user_d( ConfigManager::inst()->userTemplateDir() );
-	QStringList templates = user_d.entryList( QStringList( "*.mpt" ),
-						QDir::Files | QDir::Readable );
+	auto addTemplatesFromDir = [this]( QDir dir ) {
+		QStringList templates = dir.entryList( QStringList( "*.mpt" ),
+							QDir::Files | QDir::Readable );
 
-	m_custom_templates_count = templates.count();
-	for( QStringList::iterator it = templates.begin();
-						it != templates.end(); ++it )
-	{
-		m_templatesMenu->addAction(
-					embed::getIconPixmap( "project_file" ),
-					( *it ).left( ( *it ).length() - 4 ) );
+		if ( templates.size() && ! m_templatesMenu->actions().isEmpty() )
+		{
+			m_templatesMenu->addSeparator();
+		}
+
+		for( QStringList::iterator it = templates.begin();
+							it != templates.end(); ++it )
```

Comment:
Maybe out of scope, but I think it's a good chance to replace this with a range-based loop.